### PR TITLE
feat(nuxt): add `nitro:build:public-assets` hook

### LINKED ISSUE #14788: allow modifying public assets after build (or in hook)
### Environment

- Operating System: `Darwin`
- Node Version:     `v16.14.0`
- Nuxt Version:     `3.0.0-rc.9`
- Nitro Version:    `0.5.0`
- Package Manager:  `npm@8.19.0`
- Builder:          `vite`
- User Config:      `app`, `ssr`, `postcss`, `generate`, `nitro`, `buildModules`, `css`, `meta`, `modules`, `sitemap`, `image`, `unocss`, `publicRuntimeConfig`, `vite`, `nuxtTypedRouter`, `plugins`
- Runtime Modules:  `@unocss/nuxt@0.45.15`, `@nuxt/image-edge@1.0.0-27703611.03cc029`, `nuxt-jsonld@2.0.4`, `@funken-studio/sitemap-nuxt-3@3.2.0`
- Build Modules:    `@pinia/nuxt@0.4.1`, `@vueuse/nuxt@8.9.2`, `()`

### Reproduction

I have purgecss running from CLI after building and app. Like on the top of the build process, after updating it to rc9 purgecss (which clears vuetify styles) generate `ERR_CONTENT_LENGTH_MISMATCH` error. But I couldn't manage to reproduce it on stackblitz.

1. build the app
2. use purgecss CLI `purgecss -c purgeCSSConfig.js`
3. serve the app

### Describe the bug

Purging css after build by using purgecss breaks fetching CSS.
purgecss -c purgeCSSConfig.js

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves https://github.com/nuxt/nuxt/issues/14788

### ❓ Type of change
- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This adds a new hook (`nitro:build:assets`) to allow interacting or adding new public assets after they have been copied into `nitro.options.output.publicDir`.

Thoughts welcome on hook name (cc: @pi0).

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/3.api/4.advanced/1.hooks.md [hunk 1]
```diff
@@ -60,6 +60,7 @@ Hook                     | Arguments                  | Description
 `nitro:config`           | `nitroConfig`              | Called before initializing Nitro, allowing customization of Nitro's configuration.
 `nitro:init`             | `nitro`                    | Called after Nitro is initialized, which allows registering Nitro hooks and interacting directly with Nitro.
 `nitro:build:before`     | `nitro`                    | Called before building the Nitro instance.
+`nitro:build:assets`     | `nitro`                    | Called after copying public assets into `nitro.options.output.publicDir`. Allows modifying public assets before Nitro server is built.
```
Comment (pi0) [inline docs/3.api/4.advanced/1.hooks.md:63]: On small note, i wouln't recommand `publicDir` in docs as it might break presets depending on structure.

**Review by pi0 — APPROVED** [review]
LGTM 💯

**Comment by pi0** [comment]
Should we call it `build:public` hook? 🤔 (it is public assets)

**Comment by danielroe** [comment]
If you think that's clearer. Or maybe 'nitro:build:public-assets' if that's not too much of a mouthful?

**Comment by pi0** [comment]
I like clarity of full name 👍🏼

**Pull request merged**